metals/src/main/scala/scala/meta/internal/metals/codeactions/BracelessBracefulSwitchCodeAction.scala

```diff
@@ -0,0 +1,558 @@
+          case termIf: Term.If =>
+            Seq(
+              createCodeActionForPotentialBlockHolder(
+                termIf.thenp,
+                path,
+                document,
+                termIf.thenp,
+                "then expression"
+              ),
+              createCodeActionForPotentialBlockHolder(
+                termIf.elsep,
+                path,
+                document,
+                termIf.elsep,
+                "else expression"
+              )
+            ).flatten
+          case termFor: Term.For =>
+            createCodeActionForPotentialBlockHolder(
+              termFor,
+              path,
+              document,
+              termFor.body,
+              "for expression"
+            ).toSeq
+          case termForYield: Term.ForYield =>
+            createCodeActionForPotentialBlockHolder(
+              termForYield,
+              path,
+              document,
+              termForYield.body,
+              "yield expression"
+            ).toSeq
+          case _: Term.Match => None.toSeq
+          case termWhile: Term.While =>
+            createCodeActionForPotentialBlockHolder(
+              termWhile,
+              path,
+              document,
+              termWhile.body,
+              "while expression"
+            ).toSeq
+          case _: Defn.GivenAlias => None.toSeq
+          //          case template: Template =>
+          //            createCodeActionForTemplateHolder(
+          //              template,
+          //              path,
+          //              document,
+          //              template,
+          //              "template"
+          //            ).toSeq
+          case termBlock: Term.Block =>
+            createCodeActionForPotentialBlockHolder(
+              termBlock,
+              path,
+              document,
+              termBlock,
+              "block"
+            ).toSeq
+        }
+      }
+    result.toSeq.flatten
+  }
+
+  private def hasBraces(template: Template): Boolean = {
+    util
+      .Try(template.stats.maxBy(_.pos.end).pos.end)
+      .getOrElse(-1) != template.pos.end && util
+      .Try(template.tokens.maxBy(_.pos.end).text)
+      .getOrElse("") == "}"
+  }
+
+  private def hasBraces(termBlock: Term.Block): Boolean = {
+    util
+      .Try(termBlock.stats.maxBy(_.pos.end).pos.end)
+      .getOrElse(-1) != termBlock.pos.end && util
+      .Try(termBlock.tokens.maxBy(_.pos.end).text)
+      .getOrElse("") == "}"
+  }
+
+  /**
+   * @param blockEmbraceable the `rhs` of [[Defn.Var]] or [[Defn.Val]];
+   *                         or the `body` of [[Defn.Def]]
+   * @return whether the `rhs` or `body` passed as the
+   *         argument has braces
+   */
+  private def isBlockEmbraceableBraced(blockEmbraceable: Term): Boolean = {
+    blockEmbraceable match {
+      case termBlock: Term.Block => hasBraces(termBlock)
+      case _ => false
+    }
+  }
+
+  private def createCodeActionForCatchP(
```

Comment:
I wonder if it would be easier to add the code actions incrementally. Start with Template and Block, work on other trees in next PRs. It would be easier to review and much more focused.

Reply:
Oh, I have already covered most of them. Now the common problem in all of them is the insertion point of the open brace, and also the auto indentation before going braceless, which is super important. No one of them can be merged in isolation because they all miss the same things.